binary_search_tree/tree.py

```diff
@@ -14,35 +14,110 @@ class Tree:
     def __init__(self):
         self.root = None
 
+    def add_helper(self, current_node, key, value ):
+        if current_node == None:
+            return TreeNode(key, value)
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(current_node.right, key, value)
+        return current_node 
+
     # Time Complexity: 
     # Space Complexity: 
```

Comment:
What do you think the space and time complexity of this is? Is the time complexity dependent on the size of the binary search tree? Is there any new variables being created that grow larger depending upon the input?